### PR TITLE
tempest: Add a file to flag when a deployment is Xen-only

### PR DESCRIPTION
**Pull request by vuntz**
This will be useful to disable some tests that are failing for Xen, and
that are caused by known unfixed upstream bugs.

The check is not really thorough as we only check that there's no KVM
node, but for all practical matters, it's good enough.

### CODE REVIEW

File: chef/cookbooks/tempest/recipes/config.rb [hunk 1]
```diff
@@ -316,11 +316,17 @@
...
+xen_compute_nodes = search(:node, "roles:nova-compute-xen") || []
 docker_compute_nodes = search(:node, "roles:nova-compute-docker") || []
 
 use_resize = kvm_compute_nodes.length > 1
 use_livemigration = nova[:nova][:use_migration] && kvm_compute_nodes.length > 1
 
+if !xen_compute_nodes.empty? && kvm_compute_nodes.empty?
+  # create a flag to disable some test (lp#1443898)
+  file "#{node[:tempest][:tempest_path]}/flag-xen_only"
+end
```
Comment (vuntz) [inline chef/cookbooks/tempest/recipes/config.rb]: Ah, I forgot: do we want to remove that flag file in a `else`?

Comment (aplanas) [inline chef/cookbooks/tempest/recipes/config.rb]: IMHO yes, if we reuse the same deployment first with xen, and after that w/out xen, we will disable the test. But honestly is a corner case.

**Comment by aplanas** [comment]
+1

**Comment by rsalevsky** [comment]
+1

File: chef/cookbooks/tempest/recipes/config.rb (hunk 1 above)
Comment (toabctl) [inline chef/cookbooks/tempest/recipes/config.rb]: yes. please add a ```else```

Comment (vuntz) [inline chef/cookbooks/tempest/recipes/config.rb]: Done.

**Comment by aplanas** [comment]
+1

**Comment by rsalevsky** [comment]
Still +1
